homeassistant/bootstrap.py

```diff
@@ -151,8 +151,7 @@ async def async_setup_hass(
                 err,
             )
         else:
-            if not is_virtual_env():
-                await async_mount_local_lib_path(runtime_config.config_dir)
+            await async_mount_local_lib_path(runtime_config.config_dir)
```

Comment:
Is this not an issue anymore?

https://github.com/home-assistant/core/issues/14991

Reply:
It sort of is. But that is for dev situations. All user will have migrated a long time ago. The case could be if you accidentally run without venv, then switch to venv.

The code that clears deps is linked to a version transition that is really old (should be removed from the code)

We could do the partial change and just add root of deps.

Reply:
What is the goal of the change?

Reply:
Allow users/devs to inject custom dependant libs. Previously they used the config folder for this. But I removed that support in the referenced pull. This is to allow it but in a less error prone form.

Reply:
Won't deps in deps hide or make updates to venv deps be unreliable, as reported in the issue?

Why do we need to inject deps in the deps dirtwhen we're in a venv?

Reply:
Equally error prone in non venv docker environments. Nothing is installed here when running on haos, but it is in path. So i think we we should try to be consistent at least.

Reply:
On docker all built in integration requirements are already installed. Only custom integration requirements need to be installed on the fly. When a user upgrades a docker install, it's a new image with a new set of requirements already installed. This makes it less likely to result in problem, I think.

Reply:
Didnt know we bundled all deps on docker. So true somewhat less likely to break internal components at least during install.

When running these still take preference over the docker bundled packages, so on run you get the same risk as on venv.

Reply:
I don't think we should make this change due to the issue linked as that risk still exists.

Reply:
I think we should drop deps for the non venv case too in that case. Either you have writable site packages or a venv. Its a legacy use case i think.

We could create some new custom_dependencies to avoid some legacy issues.

@balloob thoughs?

Reply:
Hi 👋! I have made a draft introducing a `custom_packages` directory to load testing packages. https://github.com/home-assistant/core/pull/108997